src/commands/birthday/_common.ts

```diff
@@ -0,0 +1,140 @@
+import { Cursor, Collection as MongoCollection } from 'mongodb';
+import ExecutionContext from '../../ExecutionContext';
+import * as Discord from '../../Discord';
+import { Collection } from '../../database';
+
+export type Birthday = {
+  _id: number;
+  users: string[];
+};
+
+/**
+ * Calculate the day of year from a date.
+ *
+ * @param date The date to convert.
+ */
+export const calculateDayOfYear = (date: string): number => {
+  const now = new Date(date);
+  const start = new Date(now.getFullYear(), 0, 0);
+  const diff =
+    now.getTime() -
+    start.getTime() +
+    (start.getTimezoneOffset() - now.getTimezoneOffset()) * 60 * 1000;
+  const oneDay = 1000 * 60 * 60 * 24;
+  const dayOfYear = Math.floor(diff / oneDay);
+
+  return dayOfYear;
+};
+
+/**
+ * Calculate a day of year to a Date object.
+ *
+ * @param day THe day of year to convert.
+ */
+export const calculateDate = (day: number): Date => {
+  const startOfCurrentYear = new Date(new Date().getFullYear(), 0); // Initialize a date to Jan. 1 of the current year
+  const calculatedDate = new Date(startOfCurrentYear.setDate(day)); // Add the number of days.
+
+  return calculatedDate;
+};
+
+export const getTargetUser = async (
+  ctx: ExecutionContext,
+  requestorId: string | undefined,
+  targetUserId: string,
+): Promise<Discord.User | undefined> => {
+  if (targetUserId) {
```

Comment:
Isn't `targetUserId` always going to be present here?

Reply:
Updated `targetUserId` type to `string | undefined`.